cmd/gossamer/commands/root.go

```diff
@@ -0,0 +1,628 @@
+		return fmt.Errorf("failed to add --rpc-external flag: %s", err)
+	}
+
+	if err := addUint32FlagBindViper(cmd,
+		"rpc-port",
+		config.RPC.Port,
+		"HTTP-RPC server listening port",
+		"rpc.port"); err != nil {
+		return fmt.Errorf("failed to add --rpc-port flag: %s", err)
+	}
+
+	if err := addStringFlagBindViper(cmd,
+		"rpc-host",
+		config.RPC.Host,
+		"HTTP-RPC server listening hostname",
+		"rpc.host"); err != nil {
+		return fmt.Errorf("failed to add --rpc-host flag: %s", err)
+	}
+
+	if err := addStringSliceFlagBindViper(cmd,
+		"rpc-methods",
+		config.RPC.Modules,
+		"API modules to enable via HTTP-RPC, comma separated list",
+		"rpc.modules"); err != nil {
+		return fmt.Errorf("failed to add --rpc-methods flag: %s", err)
+	}
+
+	if err := addUint32FlagBindViper(cmd,
+		"ws-port",
+		config.RPC.WSPort,
+		"Websockets server listening port",
+		"rpc.ws-port"); err != nil {
+		return fmt.Errorf("failed to add --ws-port flag: %s", err)
+	}
+
+	if err := addBoolFlagBindViper(cmd,
+		"ws-external",
+		config.RPC.WSExternal,
+		"Enable external websocket connections",
+		"rpc.ws-external"); err != nil {
+		return fmt.Errorf("failed to add --ws-external flag: %s", err)
+	}
+
+	if err := addBoolFlagBindViper(cmd,
+		"unsafe-ws-external",
+		config.RPC.UnsafeWSExternal,
+		"Enable external access to websocket unsafe calls",
+		"rpc.unsafe-ws-external"); err != nil {
+		return fmt.Errorf("failed to add --ws-unsafe-external flag: %s", err)
+	}
+
+	// dummy flag to conform with the substrate cli
+	cmd.Flags().String("rpc-cors",
+		"",
+		"dummy flag to conform with the substrate cli")
+
+	return nil
+}
+
+// addCoreFlags adds core flags and binds to viper
+func addCoreFlags(cmd *cobra.Command) error {
+	cmd.Flags().StringVar(&role,
+		"role",
+		cfg.AuthorityNode.String(),
+		"Role of the node. One of 'full', 'light', or 'authority'.")
+
+	cmd.Flags().BoolVar(&validator,
+		"validator",
+		true,
+		"Run as a validator node")
+
+	if err := addBoolFlagBindViper(cmd,
+		"babe-authority",
+		config.Core.BabeAuthority,
+		"Run as a BABE authority",
+		"core.babe-authority"); err != nil {
+		return fmt.Errorf("failed to add --babe-authority flag: %s", err)
+	}
+
+	if err := addBoolFlagBindViper(cmd,
+		"grandpa-authority",
+		config.Core.GrandpaAuthority,
+		"Run as a GRANDPA authority",
+		"core.grandpa-authority"); err != nil {
+		return fmt.Errorf("failed to add --grandpa-authority flag: %s", err)
+	}
+
+	if err := addStringFlagBindViper(cmd,
+		"wasm-interpreter",
+		config.Core.WasmInterpreter,
+		"WASM interpreter",
+		"core.wasm-interpreter"); err != nil {
+		return fmt.Errorf("failed to add --wasm-interpreter flag: %s", err)
+	}
+
+	if err := addDurationFlagBindViper(cmd,
+		"grandpa-interval",
+		config.Core.GrandpaInterval,
+		"GRANDPA voting period in seconds",
+		"core.grandpa-interval"); err != nil {
+		return fmt.Errorf("failed to add --grandpa-interval flag: %s", err)
+	}
+
+	if err := addBoolFlagBindViper(cmd,
```

Comment:
should this be removed?

Reply:
I'll remove it after merging `development`